### PR TITLE
Made some stylistic changes to `apply_dp`

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #449
* #448
* #447
* __->__ #446
* #445
* #444

### CODE REVIEW

File: torchtitan/parallelisms/parallelize_llama.py [hunk 1]
```diff
@@ -458,23 +458,21 @@ def apply_dp(model, world_mesh, parallel_dims, job_config: JobConfig):
...
+        else:
+            # As an optimization, do not reshard after forward for the last
+            # transformer block since FSDP would prefetch it immediately
+            reshard_after_forward = int(layer_id) < len(model.layers) - 1
         fully_shard(
             transformer_block,
             **fsdp_config,
             reshard_after_forward=reshard_after_forward,
         )
-        model.layers[layer_id] = transformer_block
```
Comment (awgu) [inline torchtitan/parallelisms/parallelize_llama.py:476]: do not need to assign back in

File: torchtitan/parallelisms/parallelize_llama.py [hunk 2]
```diff
@@ -458,23 +458,21 @@ def apply_dp(model, world_mesh, parallel_dims, job_config: JobConfig):
...
-
-    model = fully_shard(
```
Comment (awgu) [inline torchtitan/parallelisms/parallelize_llama.py:478]: ditto

File: torchtitan/parallelisms/parallelize_llama.py [hunk 3]
```diff
@@ -458,23 +458,21 @@ def apply_dp(model, world_mesh, parallel_dims, job_config: JobConfig):
...
-
     for layer_id, transformer_block in model.layers.items():
-        # As an optimization, do not reshard after forward for the last
-        # transformer block since FSDP would prefetch it immediately.
-        # When using Pipeline Parallelism, generally zero-2 is best so as to avoid repeated reshardings
-        # per microbatch.
-        reshard_after_forward = (
-            int(layer_id) < len(model.layers) - 1 and not parallel_dims.pp_enabled
-        )
+        if parallel_dims.pp_enabled:
```
Comment (awgu) [inline torchtitan/parallelisms/parallelize_llama.py:464]: separate out the comments to make it clearer

**Review by tianyu-l — APPROVED** [review]
nice!

**Pull request merged**